adr/2025_06_16-nested-blocks.md

```diff
@@ -0,0 +1,155 @@
+# BlockNote Nested Blocks
+
+There are two separate problems when it comes to "nested blocks" support in BlockNote:
+
+- **nested-blocks** The first is the ability for a block to contain other blocks within it (e.g. a table cell having not just inline content, but actual other blocks inside it)
+- **content-fields** The second is the ability for a block to contain multiple pieces of content within it (e.g. an alert block having a title & description field (which contain inline content))
+
+Let's start with the first problem, nested blocks. By describing existing block relationships:
+
+## Block with inline content
+
+This is the simplest case, and the only one that is currently supported by BlockNote.
+
+```json
```

Comment:
Worth mentioning that the current inline content object shape is a mess. You have `StyledText` which is a different shape to `Link`, which is also different to custom inline content. This makes trying to do basic things like reading plain text kind of a nightmare, and is something that we definitely need to address.

Reply:
Good thing to add, I don't have an answer for this atm. But feel free to write a proposal